src/renderer/components/PromotionModal.vue

```diff
@@ -0,0 +1,134 @@
+<template>
+  <div
+    class="prom-container"
+  >
+    <div
+      id="op1"
+      class="prom-option"
+      :class="[ turn ? 'promOptionW1' : 'promOptionB4']"
+      @click="close('1')"
+    />
+    <div
+      id="op2"
+      class="prom-option"
+      :class="[ turn ? 'promOptionW2' : 'promOptionB3']"
+      @click="close('2')"
+    />
+    <div
+      id="op3"
+      class="prom-option"
+      :class="[ turn ? 'promOptionW3' : 'promOptionB2']"
+      @click="close('3')"
+    />
+    <div
+      id="op4"
+      class="prom-option"
+      :class="[ turn ? 'promOptionW4' : 'promOptionB1']"
+      @click="close('4')"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'PromotionModal',
+  data () {
+    return {
+      promDir: { 1: 'q', 2: 'n', 3: 'r', 4: 'b' }
+    }
+  },
+  computed: {
+    pieceStyle () {
+      return this.$store.getters.pieceStyle
+    },
+    turn () {
+      if (this.$store.getters.turn === undefined) {
```

Comment:
I think undefined would be considered false in a condition so this check is not really necessary.